sources/updater/updog/src/main.rs

```diff
@@ -424,6 +441,24 @@ fn initiate_reboot() -> Result<()> {
     Ok(())
 }
 
+// Returns the target version. force_version takes priority. If that's not specified, take it from the config.
+fn get_target_version(config: &Config, force_version: &Option<Version>) -> Result<Option<Version>> {
+    match force_version {
+        Some(version) => Ok(Some(version.to_owned())),
+        // If no version is specified on the commandline, get the version from the config
+        None => {
+            if config.version_lock != "latest" {
+                match FriendlyVersion::try_from(config.version_lock.as_str()) {
+                    Ok(version) => Ok(Some(version.try_into().context(error::BadVersion)?)),
+                    Err(_) => Ok(None),
```

Comment:
I'm having a hard time following this function (can we make it a bit more linear?) but I think this line is saying that if we have an invalid configured version lock, we'll be unlocked?  I think we want that to fail instead?